### DISCUSSION TITLE
why is virtualenv the default virtualenv and not python's venv

### DISCUSSION THREAD

**Discussion by deronnax** [pdm-project/pdm#3268] [General]
Why not use python's official virtualenv as PDM default virtualenv backend?

**Comment by frostming** [comment]
have you used a python distribution that needs to install python-venv module separately?

**Reply by deronnax** [reply]
A while ago, when PDM wasn't out yet. But yeah OK, I see.

**Reply by deronnax** [reply]
For the people wondering the why, some distributions like Ubuntu/Debian do what is called "unbundling": they separate pip, virtualenv (and setuptools when applicable [^1]) from the shipped python, which means you can find yourself on a systems where there is incomplete python. You are not supposed to use directly the python provided by your linux distribution, but most people do it, especially beginners.

[^1]: setuptools was removed from python starting on python 3.12.

**Comment by deronnax** [comment]
@frostming could we have a few lines about `virtualenv` vs `venv`, for the people to make a choice? For me, I have no idea about virtualenv vs venv.

**Reply by frostming** [reply]
In my opinion, if they don't understand the difference between these two backends, they don't need to. We offer the ability to switch backends to those who are aware of this difference.

Not understanding the differences doesn't affect using PDM.
